### DISCUSSION TITLE
how to manage gpu device in k8s cluster with prefect

### DISCUSSION THREAD

**Discussion by jsshizhan** [PrefectHQ/prefect#16137] [Q&A]
### Bug summary

 100 flow run,k8s cluster have 10 gpu device ,one gpu device run a flow run,how to manage the gpu device and flow run

### Version info

```Text
prefect version 3.0.11
python version 3.12
```

**Comment by desertaxle** [comment]
Hey @jsshizhan! I converted this to a discussion since it seemed more like a question than a bug report.

If you want to control the nodes that your flows run on, you can use [Kubernetes node labels and selectors](https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/). When you use a Kubernetes worker, each flow run is executed in a separate Kubernetes job. You can control the configuration of that created job via the base job template of the work pool from which your worker is pulling work.

For example, if you have your GPU nodes labeled a type GPU:
```
kubectl label nodes <your-node-name> type=gpu
```
You could add the following to your work pool's base job template to only run on those nodes:
```json
{
    ...
    "job_configuration":
        ...
        "spec": {
            ...
             "nodeSelector": {
                 "type": "gpu"
             }
        }
    }
}
```
If you want to make sure that only one flow runs on each node at a time, you can use pod labels to implement [anti-affinity between pods](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#types-of-inter-pod-affinity-and-anti-affinity). Once again, you can customize this via the base job template for your work pool.

Let me know if you have any other questions!

**Reply by jsshizhan** [reply]
@desertaxle if i have 100 flow_runs,2nodes,every node have 4gpu devices,1gpu device must run 1flow_run,how to efficient arrangement late state flow_runs?
